src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -117,15 +117,19 @@ ss::future<> service::do_start() {
 }
 
 ss::future<> service::create_internal_topic() {
-    auto replication_factor = _config.schema_registry_replication_factor();
+    // Use the default topic replica count, unless our specific setting
+    // for the schema registry chooses to override it.
+    int16_t replication_factor = _config.schema_registry_replication_factor();
+    if (replication_factor == -1) {
+        replication_factor
+          = config::shard_local_cfg().default_topic_replication();
+    }
```

Comment:
This section is primarily for logging, since sending -1 is allowed and does the right thing.

I think the non-rpk changes in this commit were intended to land in the previous one?

Reply:
Fixed the mixed commits.

The -1 thing is in there because it was meant to be a std::optional but had to back that out because of YAML parsing weirdness, so I'll leave this as-is until we can come back later and fix that (https://github.com/vectorizedio/redpanda/issues/1914)